lib/Db/CoreQueryBuilder.php

```diff
@@ -1156,13 +1157,36 @@ public function leftJoinInitiator(
 			)
 		);
 
+
+		$listMembershipCircleAlias = [$aliasMembership];
+		if ($this->getBool('initiatorDirectMember', $options, false)) {
+			try {
+				$aliasDirectInitiator = $this->generateAlias($alias, self::DIRECT_INITIATOR, $options);
+				$listMembershipCircleAlias[] = $aliasDirectInitiator;
+			} catch (RequestBuilderException $e) {
```

Comment:
I would prefer at least a code comment why an Exception is ignored even if it is a valid use case.

Reply:
In fact this Exception have to be ignored.

There is some method in this class that are cross-called to generate the result.
As an example, a `member` belongs to a `circle` that have an `member` as owner that is based on a `circle`. You can see the redundancy here when retrieving a members list of a circle.
This is an example, but the idea is that `generateAlias()` will check [self::$SQL_PATH](https://github.com/nextcloud/circles/blob/0e1cab95a16b3abb9545265443724e82f9cbc65b/lib/Db/CoreQueryBuilder.php#L85) to build the request and avoid infinite loop.

Should I still comment in the code ?

Reply:
You shall not comment in/out the code, but add a comment why exactly this exception is catched

Reply:
Yes, please mention it in the code.